### PR TITLE
Support Prometheus UNIT metadata

### PR DESCRIPTION
**Pull request by reyang**
## Changes

* Added support for UNIT metadata.
* Renamed some of the internal methods to better align with OpenMetrics terms (e.g. "metadata").
* Adjusted the ordering of the metadata to follow the OpenMetrics best practices.

Reference: https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#overall-structure

image

For significant contributions please make sure you have completed the following items:

* [x] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusSerializerTests.cs [hunk 1]
```diff
@@ -96,6 +96,34 @@ public void GaugeZeroDimensionWithUnit()
...
+            var buffer = new byte[85000];
+            var metrics = new List<Metric>();
+
+            using var meter = new Meter(Utils.GetCurrentMethodName());
+            using var provider = Sdk.CreateMeterProviderBuilder()
+                .AddMeter(meter.Name)
+                .AddInMemoryExporter(metrics)
+                .Build();
+
+            meter.CreateObservableGauge("test_gauge", () => 123, unit: "seconds", description: "Hello, world!");
```
Comment (cijothomas) [inline test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusSerializerTests.cs:117]: nice description :)

**Review by CodeBlanch — APPROVED** [review]
LGTM

File: test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusSerializerTests.cs (hunk 1 above)
Comment (CodeBlanch) [reply test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusSerializerTests.cs:117]: Needs some clever Greek mythology thing. Defer to @alanwest to come up with it 🤣

Comment (alanwest) [reply test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusSerializerTests.cs:117]: I mean, would "Hello, Prometheus!" not have been clever enough? And also what the heck do I know about Greek mythology?

Comment (reyang) [reply test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusSerializerTests.cs:117]: Do we consider the Prometheus Exporters to be renamed to OpenMetrics Exporters? 😆

**Pull request merged**